### PR TITLE
build: Decrease core build size

### LINKED ISSUE #81: [BUILD] dependencies needed for functime
In #72 I was looking at functime package size, which is as much as 1GB. I was wondering what dependencies where actually required:

https://github.com/neocortexdb/functime/blob/5480874bb8ff968726955ace6b71d0be36b64fe4/pyproject.toml#L29-L46

- kaleido is used for plots
- pynndescent seems unused
- rich requests tabulate fastapi seem unused too

Should some of those be removed?

### PR DESCRIPTION
**Pull request by topher-lo**
Removes;
- pynndescent
- rich

Makes optional:
- plotly 
- kaleido

Resolves #81 #72

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -32,12 +32,8 @@ dependencies = [
     "holidays",
     "kaleido==0.2.1",
```
Comment (baggiponte) [inline pyproject.toml:33]: I think you might have to remote this as well

File: pyproject.toml [hunk 2]
```diff
@@ -50,10 +46,12 @@ dependencies = [
 Homepage = "https://github.com/descendant-ai/functime"
 
 [project.optional-dependencies]
+ann = ["pylance"]
 cat = ["catboost"]
 xgb = ["xgboost"]
 lgb = ["lightgbm"]
 llm = ["openai", "tabulate", "tenacity", "tiktoken"]
+plot = ["plotly", "kaleido"]
```
Comment (baggiponte) [inline pyproject.toml:54]: Perhaps you should write `"kaleido==0.2.1"`?

Comment (topher-lo) [reply pyproject.toml]: 👍

**Comment by baggiponte** [comment]
Currently this brings core functime from ~1GB to some 530MB.

**Comment by baggiponte** [comment]
Uh I just realised that `requests` might belong to the `test` optional dependencies block rather than the main ones.

**Pull request merged**